### PR TITLE
Bamtools split wrapper.

### PR DESCRIPTION
**Pull request by Smeds**

### CODE REVIEW

File: bio/bamtools/split/test/Snakefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+rule bamtools_split:
+    input:
+        "mapped/{sample}.bam",
+    output:
+        "mapped/{sample}.REF_xx.bam",
+    params:
+        #  -reference is the default value and can be left out, could also be '-paired' and '-tag <tag_name>'
+        split_type="-reference",
```
Comment (jafors) [inline bio/bamtools/split/test/Snakefile:8]: Could you add an `extra` field here, if someone wants to use other flags like `-refPrefix`?

```suggestion
    params:
        #  -reference is the default value and can be left out, could also be '-paired' and '-tag <tag_name>'
        split_type="-reference",
        extra=""
```

File: bio/bamtools/split/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+__author__ = "Patrik Smeds"
+__copyright__ = "Copyright 2021, Patrik Smeds"
+__email__ = "koester@jimmy.harvard.edu"
```
Comment (jafors) [inline bio/bamtools/split/wrapper.py:3]: You can give your own mail address here. ;)

File: bio/bamtools/split/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from snakemake.shell import shell
+
+log = snakemake.log_fmt_shell(stdout=False, stderr=True)
+
+split_type = snakemake.params.get("split_type", "-reference")
+
+if len(snakemake.input) != 1:
+    raise ValueError("One bam input file expected, got: " + str(len(snakemake.input)))
+
+shell("bamtools split -in {snakemake.input} " + split_type + " {log}")
```
Comment (jafors) [inline bio/bamtools/split/wrapper.py:16]:
```suggestion
shell("bamtools split -in {snakemake.input} {split_type} {log}")
```

**Review by jafors — CHANGES_REQUESTED** [review]
Thanks for the work, this should be a valuable addition! Find some small remarks below.

File: bio/bamtools/split/wrapper.py (hunk 2 above)
Comment (Smeds) [reply bio/bamtools/split/wrapper.py]: Done

File: bio/bamtools/split/test/Snakefile (hunk 1 above)
Comment (Smeds) [reply bio/bamtools/split/test/Snakefile]: Changed split_type to extra

File: bio/bamtools/split/wrapper.py (hunk 3 above)
Comment (Smeds) [reply bio/bamtools/split/wrapper.py]: Changed split_type to extra

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**